### PR TITLE
Support for "SyncPWM" and "8XPWM" (see http://git.io/p2tz)

### PR DESCRIPTION
**Pull request by fiendie**
Adds for support for 1/8th PWM pulse length aliases as described by @sim- here: http://git.io/p2tz
Big thanks to @luggi ;)

### CODE REVIEW

File: src/drv_pwm.c [hunk 1]
```diff
@@ -187,19 +189,19 @@ static void pwmOCConfig(TIM_TypeDef *tim, uint8_t channel, uint16_t value)
...
             break;
         case TIM_Channel_3:
             TIM_OC3Init(tim, &TIM_OCInitStructure);
-            TIM_OC3PreloadConfig(tim, TIM_OCPreload_Enable);
+            TIM_OC3PreloadConfig(tim, TIM_OCPreload_Disable);
             break;
         case TIM_Channel_4:
             TIM_OC4Init(tim, &TIM_OCInitStructure);
-            TIM_OC4PreloadConfig(tim, TIM_OCPreload_Enable);
+            TIM_OC4PreloadConfig(tim, TIM_OCPreload_Disable);
```
Comment (trollcop) [inline src/drv_pwm.c]: these 4 changes are not explained

Comment (trollcop) [inline src/drv_pwm.c]: these 4 changes are not explained

Comment (trollcop) [inline src/drv_pwm.c]: fuck shithub for breaking the comments

Comment (luggi) [inline src/drv_pwm.c]: this to disable the ouput compare register buffering, it only gets updated on a timer overflow i think, which doesn't happen with syncpwm disabled

Comment (trollcop) [inline src/drv_pwm.c]: uh, and what happens when this useless pwm is NOT enabled?

Comment (luggi) [inline src/drv_pwm.c]: you are right, in rare cases, it could cause the output wrong values.
@fiendie will update the pr

**Comment by fiendie** [comment]
Should've tested that build script... 
It builds fine though. The latest commit should fix TIM_OCPreload behaviour.

**Comment by fiendie** [comment]
ok to test

File: src/board.h [hunk 2]
```diff
@@ -96,6 +96,8 @@ typedef enum {
     FEATURE_VARIO = 1 << 13,
     FEATURE_3D = 1 << 14,
     FEATURE_FW_FAILSAFE_RTH = 1 << 15,
+    FEATURE_SYNCPWM = 1 << 16,
+    FEATURE_FASTPWM = 1 << 17
```
Comment (trollcop) [inline src/board.h]: add comma for next change not gaying up the existing line

File: src/drv_pwm.c [hunk 3]
```diff
@@ -184,22 +187,29 @@ static void pwmOCConfig(TIM_TypeDef *tim, uint8_t channel, uint16_t value)
     TIM_OCInitStructure.TIM_OCPolarity = TIM_OCPolarity_Low;
     TIM_OCInitStructure.TIM_OCIdleState = TIM_OCIdleState_Set;
```
Comment (trollcop) [inline src/drv_pwm.c]: why space?

File: src/drv_pwm.c [hunk 4]
```diff
@@ -403,13 +427,22 @@ bool pwmInit(drv_pwm_config_t *init)
...
+            if (init->motorPwmRate > 500 || init->fastPWM)
+                mhz = PWM_TIMER_8_MHZ;
             else
                 mhz = PWM_TIMER_MHZ;
+
             hz = mhz * 1000000;
 
-            motors[numMotors++] = pwmOutConfig(port, mhz, hz / init->motorPwmRate, init->idlePulse);
+            if (init->syncPWM)
+                period = 8000 * mhz; // 8ms period in syncPWM mode, cycletime should be smaller than this
```
Comment (trollcop) [inline src/drv_pwm.c]: What does this mean?
Can a user choose something that will fuck shit up?

**Pull request merged**